pkg/operator/controller/canary/controller.go

```diff
@@ -110,14 +112,53 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 			if !ok {
 				return false
 			}
-			// if Spec.Port has changed, do not trigger a reconcile
-			return cmp.Equal(oldRoute.Spec.Port, newRoute.Spec.Port)
+
+			// Check if the port in newRoute is one of the ports exposed by the canary service.
+			isValidPort := false
+			for _, portNum := range CanaryPorts {
+				switch newRoute.Spec.Port.TargetPort.Type {
+				case intstr.Int:
+					if newRoute.Spec.Port.TargetPort.IntVal == portNum {
+						isValidPort = true
+						break
+					}
+				case intstr.String:
+					// Not currently supported; for simplicity, the canary check doesn't do name to port translations,
+					// so if the canary's target port is a port name, we want to treat it as an invalid port and
+					// reconcile the route.
+				}
+			}
+
+			// If newRoute's port isn't one of the canary service ports, reconcile the route.
+			if !isValidPort {
+				return true
+			}
+
+			// If newRoute's port is one of the canary service ports, the change may just be the ingress operator
+			// rotating canary ports. If the *only* change between oldRoute and newRoute is the port change, skip
+			// reconciling.
+			oldRoute.Spec.Port = newRoute.Spec.Port
```

Comment:
Ah, the assignment is so that the `cmp.Equal` effectively ignores a difference in `Spec.Port`, right?  Using something like `cmp.Equal(oldRoute.Spec, newRoute.Spec, cmpopts.IgnoreFields(routev1.RouteSpec{}, "Port"))` would be a little more obvious.